toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -1392,6 +1497,44 @@ func formatCycleErrorMessage(cycle []*PkgNode, err error) error {
 	return fmt.Errorf("cycles detected in dependency graph")
 }
 
+// nodesProvidedBySRPM returns all RPMs produced from a SRPM file.
+// could also return []*PkgNode
+// instead of map[PkgNode.VersionedPkg]bool
+// refer AllRunNodes
+func nodesProvidedBySRPM(srpmPath string, pkgGraph *PkgGraph, graphMutex *sync.RWMutex) (rpmNodes map[*pkgjson.PackageVer]bool) {
```

Comment:
Could this be public and return lookup nodes? I've got some code in the works that does something quite similar and I'd love to just use the same function.

Reply:
Done! the function is made public (NodesProvidedBySRPM) and it now returns pkgNodes.